src/meta/Drag.ts

```diff
@@ -0,0 +1,182 @@
+import { deepAssign } from '@dojo/core/lang';
+import global from '@dojo/shim/global';
+import { assign } from '@dojo/shim/object';
+import WeakMap from '@dojo/shim/WeakMap';
+import { Base } from './Base';
+
+export interface DragResults {
+	/**
+	 * The movement of pointer during the duration of the drag state
+	 */
+	delta: Position;
+
+	/**
+	 * Is the DOM node currently in a drag state
+	 */
+	isDragging: boolean;
+}
+
+interface NodeData {
+	dragResults: DragResults;
+	invalidate: () => void;
+	last: Position;
+	start: Position;
+}
+
+export interface Position {
+	x: number;
+	y: number;
+}
+
+function createNodeData(invalidate: () => void): NodeData {
+	return {
+		dragResults: deepAssign({}, emptyResults),
+		invalidate,
+		last: createPosition(),
+		start: createPosition()
+	};
+}
+
+/**
+ * Creates an empty position
+ */
+function createPosition(): Position {
+	return { x: 0, y: 0 };
+}
+
+/**
+ * A frozen empty result object, frozen to ensure that no one downstream modifies it
+ */
+const emptyResults = Object.freeze({
+	delta: Object.freeze(createPosition()),
+	isDragging: false
+});
+
+/**
+ * Return the x/y position for an event
+ * @param event The pointer event
+ */
+function getPosition(event: PointerEvent): Position {
+	return {
+		x: event.pageX,
+		y: event.pageY
+	};
+}
+
+/**
+ * Return the delta position between two positions
+ * @param start The first position
+ * @param current The second position
+ */
+function getDelta(start: Position, current: Position): Position {
+	return {
+		x: current.x - start.x,
+		y: current.y - start.y
+	};
+}
+
+class DragController {
+	private _nodeMap = new WeakMap<HTMLElement, NodeData>();
+	private _dragging: HTMLElement | undefined = undefined;
```

Comment:
It could be, but considering I was explicitly assigning `undefined`, the property will always be on the instance.  Felt more accurate to not be optional.